### PR TITLE
Stop refetch when status is not pending 

### PR DESCRIPTION
**Pull request by ainouzgali**
### What kind of change does this PR introduce? 

- [ ] Bug
- [x] Feature
- [ ] Docs
- [ ] Other(s)

### CODE REVIEW

**Comment by ainouzgali** [comment]
https://www.loom.com/share/df8e110be41e485cbb50e43bd72a2be0

File: apps/web/src/components/activity/ExecutionDetailsModal.tsx [hunk 1]
```diff
@@ -20,11 +23,20 @@ export function ExecutionDetailsModal({
   onViewDigestExecution?: (digestNotificationId: string) => void;
 }) {
   const theme = useMantineTheme();
+  const [shouldRefetch, setShouldRefetch] = useState(true);
```
Comment (scopsy) [inline apps/web/src/components/activity/ExecutionDetailsModal.tsx:26]:
```suggestion
  const shouldRefetch = status && status !== ExecutionDetailsStatusEnum.PENDING;
```

I think we can actually remove this state as I think it's not really needed and makes the code a bit easier to read :)

**Review by scopsy — APPROVED** [review]
@ainouzgali looks good, i would just remove the extra `useState` and just do a variable for it

File: apps/web/src/components/activity/ExecutionDetailsModal.tsx (hunk 1 above)
Comment (ainouzgali) [reply apps/web/src/components/activity/ExecutionDetailsModal.tsx:26]: My problem with that was that I don't actually have the status until after the first run of the query. As I need the response to get the status

Comment (scopsy) [reply apps/web/src/components/activity/ExecutionDetailsModal.tsx:26]: Oh got ya, so you need the default here to initially fetch it. 🤞

**Pull request merged**